pkg/controller/operators/olm/requirements.go

```diff
@@ -358,6 +411,59 @@ func (a *Operator) permissionStatus(strategyDetailsDeployment *v1alpha1.Strategy
 	return permMet && clusterPermMet, statuses, nil
 }
 
+func permissionsPreviouslyCreated[T, U metav1.Object](
+	permission v1alpha1.StrategyDeploymentPermissions,
+	csv *v1alpha1.ClusterServiceVersion,
+	listRoles func(labels.Selector) ([]T, error),
+	listBindings func(labels.Selector) ([]U, error),
+) (bool, error) {
+	// first, find the (cluster)role
+	ruleHash, err := resolver.PolicyRuleHashLabelValue(permission.Rules)
+	if err != nil {
+		return false, fmt.Errorf("failed to hash permission rules: %w", err)
+	}
+	roleSelectorMap := ownerutil.OwnerLabel(csv, v1alpha1.ClusterServiceVersionKind)
+	roleSelectorMap[resolver.ContentHashLabelKey] = ruleHash
+	roleSelectorSet := labels.Set{}
+	for key, value := range roleSelectorMap {
+		roleSelectorSet[key] = value
+	}
+	roleSelector := labels.SelectorFromSet(roleSelectorSet)
+	roles, err := listRoles(roleSelector)
+	if err != nil {
+		return false, err
+	}
+
+	if len(roles) == 0 {
+		return false, nil
+	}
+
+	// then, find the (cluster)rolebinding, if we found the role
+	bindingHash, err := resolver.RoleReferenceAndSubjectHashLabelValue(rbacv1.RoleRef{
+		Kind:     "Role",
+		Name:     roles[0].GetName(),
```

Comment:
We only ever expect 1, or ?

Reply:
Given that one of our labels in the selector is the hash of the rules, we do just expect the one. The workqueue will dedupe keys so I'd expect one worker in the controller to be creating RBAC, not many - so I don't know how we'd race with ourselves to create more than one instance of the role?